spec/models/manageiq/providers/microsoft/infra_manager/vm_scan_spec.rb

```diff
@@ -1,20 +1,99 @@
 describe VmScan do
-  context "A single VM Scan Job on Microsoft provider" do
-    let(:vm) do
-      vm = double("ManageIQ::Providers::Microsoft::InfraManager::Vm")
-      allow(vm).to receive(:kind_of?).with(ManageIQ::Providers::Openstack::CloudManager::Vm).and_return(false)
-      allow(vm).to receive(:kind_of?).with(ManageIQ::Providers::Microsoft::InfraManager::Vm).and_return(true)
-      vm
-    end
-    let(:job) { VmScan.new(:context => {}, :options => {}) }
-
-    describe "#call_snapshot_create" do
-      it "executes VmScan#create_snapshot and send signal :snapshot_complete" do
-        allow(VmOrTemplate).to receive(:find).and_return(vm)
-        expect(job).to receive(:create_snapshot).and_return(true)
-        expect(job).to receive(:signal).with(:snapshot_complete)
-        job.call_snapshot_create
+  let(:miq_server) { FactoryBot.create(:miq_server) }
+  let(:ems)  { FactoryBot.create(:ems_microsoft) }
+  let(:vm)   { FactoryBot.create(:vm_microsoft, :ext_management_system => ems) }
+  let(:job)  { VmScan.create_job }
+  let(:snapshot_description) { "Snapshot description" }
+
+  before do
+    allow(VmOrTemplate).to receive(:find).and_return(vm)
```

Comment:
Can you initialize `Job` with `:target_id => vm.id` so that you can be more specific here and add a `.with(vm.id)`?

I tried to delete this completely but it does make it harder to stub out the provider calls later on

Reply:
Done!